client/demo/ex/assignment2/expensesController.js

```diff
@@ -0,0 +1,101 @@
+angular
+    .module('expensesApp')
+    .controller('ExpenseController', ExpenseController);
+
+
+ExpenseController.$inject = ['$filter'];
+
+function ExpenseController($filter){
+    vm = this;
+
+    vm.expenses = [];
+    vm.expense = {};
+    vm.types = ['food', 'transportation', 'lodging', 'financial', 'sales', 'other.'];
+    vm.isEditing = false;
+
+    vm.total = 0;
+
+    vm.newExpense = function(){
```

Comment:
Please avoid anonymous funtoins use named functions instead and assign to controller
``
vm.newExpense = newExpense;
function newExpense(){ 
}
``